### PR TITLE
Models

### LINKED ISSUE #5: Models

### PR DESCRIPTION
**Pull request by lazarnagulov**
closes #5

### CODE REVIEW

File: PetCenter/PetCenter/Domain/Model/Account.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+namespace PetCenter.Domain.Model
+{
+    public class Account(string username, string email, string password)
+    {
+        public Guid Id { get; } = Guid.NewGuid();
+        public string Username { get; set; } = username;
+        public string Email { get; set; } = email;
+        public string Password { get; set; } = password;
+        public Person? Person { get; set; }
+    }
```
Comment (FilipT03) [inline PetCenter/PetCenter/Domain/Model/Account.cs]: Account should have an AccountType.

File: PetCenter/PetCenter/Domain/Model/AnimalType.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PetCenter.Domain.Model
+{
+    public class AnimalType(string type, string race)
+    {
+        public string Type { get; set; } = type;
+        public string Race { get; set; } = race;
```
Comment (FilipT03) [inline PetCenter/PetCenter/Domain/Model/AnimalType.cs]: Breed instead of Race.

File: PetCenter/PetCenter/Domain/Model/Account.cs (hunk 1 above)
Comment (lazarnagulov) [reply PetCenter/PetCenter/Domain/Model/Account.cs]: Fixed.

File: PetCenter/PetCenter/Domain/Model/AnimalType.cs (hunk 2 above)
Comment (lazarnagulov) [reply PetCenter/PetCenter/Domain/Model/AnimalType.cs]: Fixed.

File: PetCenter/PetCenter/Domain/Model/Post.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        private readonly List<Offer> _offers = [];
+        private readonly List<Comment> _comments = [];
+
+        public void AddOffer(Offer offer) => _offers.Add(offer);
+        public void RemoveOffer(Offer offer) => _offers.Remove(offer);
+        public void AddLike(Person person) => _likes.Add(person);
+        public void RemoveLike(Person person) => _likes.Remove(person);
+        public void AddComment(Comment comment) => _comments.Add(comment);
+        public void RemoveComment(Comment comment) => _comments.Remove(comment);
+    }
```
Comment (FilipT03) [inline PetCenter/PetCenter/Domain/Model/Post.cs]: Missing post date.

File: PetCenter/PetCenter/Domain/Model/Review.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PetCenter.Domain.Model
+{
+    public class Review(Offer offer, int grade, string comment)
+    {
+        public Offer Offer { get; set; } = offer;
```
Comment (FilipT03) [inline PetCenter/PetCenter/Domain/Model/Review.cs]: The review doesn't need to see the offer.

File: PetCenter/PetCenter/Domain/Model/Post.cs (hunk 3 above)
Comment (lazarnagulov) [reply PetCenter/PetCenter/Domain/Model/Post.cs]: Fixed.

File: PetCenter/PetCenter/Domain/Model/Review.cs (hunk 4 above)
Comment (lazarnagulov) [reply PetCenter/PetCenter/Domain/Model/Review.cs]: Fixed.

**Review by FilipT03 — APPROVED** [review]
Excellent!

**Pull request merged**